Sources/Swift/Helper/SentryDataWrapper.swift

```diff
@@ -0,0 +1,353 @@
+//
+//  SentryDataWrapper.swift
+//  Sentry
+//
+//  Created by Philip Niedertscheider on 12.12.24.
+//  Copyright © 2024 Sentry. All rights reserved.
+//
+
+/// A drop-in replacement for the standard ``Swift.Data`` but with automatic tracking for file I/O operations.
+///
+/// This structure is intended to resemble the same method signatures as of ``Swift.Data``.
+@available(iOS 18, macOS 15, tvOS 18, *)
+@frozen public struct SentryDataWrapper: Equatable, Hashable, RandomAccessCollection, MutableCollection, RangeReplaceableCollection, MutableDataProtocol, ContiguousBytes, Sendable {
```

Comment:
l: Name too long IMO.

Suggestions
- SentryData
- TrackedData
- TracedData

Reply:
"Clarity is more important than brevity" so I don't take issue with a difference of a couple characters. 

We currently have `SentryNSDataTracker`, I would suggest a parallel convention of `SentrySwiftDataTracker`, with the _annoying exception_ of there now being a SwiftData product offering from Apple that would create a naming overload ಠ_ಠ Maybe just `SentryDataTracker`. Even though the namespacing isn't required in Swift, I still think it's a good idea to keep `Sentry...` tacked onto our types.

Reply:
`SentryNSDataTracker` will be renamed to `SentryFileIOTracker` in #4634, rather than a helper class to deal with spans, so it might be used by `SentryDataWrapper`, but it does not serve the same purpose.

I like `Sentry` in the name, but dislike the keyword `Wrapper`, but not due to the length. Therefore I would also agree with `SentryTrackedData` or similar